apis/cloudwatchlogs/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// CustomLogGroupParameters contains the additional fields for LogGroup.
+type CustomLogGroupParameters struct {
+	// The number of days to retain the log events in the specified log group.
+	// To set a log group to never have log events expire, use DeleteRetentionPolicy
```

Comment:
see `0` is added as a special case for crossplane. Not sure if we have something similar elsewhere. I guess `nil` causes issues with late init? Another option might be to use k8s `intstr` and support `"Forever"` or `"Infinite"`?

I don't think this comment relevant in user facing docs exactly as it is written here: `// To set a log group to never have log events expire, use DeleteRetentionPolicy`, but I do think it would be good to have a note about the special case w/ `0` and that this calls `DeleteRetentionPolicy`.

Reply:
Fair enough i will Change the Description - my Idea was to use 0 Like Cloudwatch LogGroup in Terraform "If you select 0, the events in the log group are always retained and never expirex

Reply:
changed the description